File: kafka-test-application/src/main/java/com/mycompany/kafka/KafkaConsumerManager.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.kafka;

import java.util.Arrays;
import java.util.Properties;
import org.apache.kafka.clients.consumer.ConsumerRecord;
import org.apache.kafka.clients.consumer.ConsumerRecords;
import org.apache.kafka.clients.consumer.KafkaConsumer;

/**
 *
 * @author vagrant
 */
public class KafkaConsumerManager {

    private KafkaConsumer<String, String> consumer;
    
    private static final String KAFKASERVER = "192.168.22.10:9092";
    //private static final String ZOOKEEPER = "192.168.22.10:2181";

    public KafkaConsumerManager() {
        Properties props = new Properties();
        props.put("bootstrap.servers", KAFKASERVER);
        props.put("group.id", "testGroup");
        props.put("enable.auto.commit", "true");
        props.put("auto.commit.interval.ms", "1000");
        props.put("session.timeout.ms", "30000");
        props.put("key.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");
        props.put("value.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");
//        props.put("broker.id", 0);
//        props.put("zookeeper.connect", ZOOKEEPER);
        
        consumer = new KafkaConsumer<>(props);
    }

    public void start() {
        System.out.println("Start Consumer");
        consumer.subscribe(Arrays.asList("test"));
        while (true) {
            ConsumerRecords<String, String> records = consumer.poll(100);
            for (ConsumerRecord<String, String> record : records) {
                System.out.printf("Consumer: offset = %d, key = %s, value = %s \n", record.offset(), record.key(), record.value());
            }
        }
    }
}

File: kafka-cloudera/src/main/scala-2.11/KafkaConsumerManager.scala
import java.util.{Arrays, Properties}

import org.apache.kafka.clients.consumer.KafkaConsumer

/**
  * Created by Lorenz on 20/04/2016.
  */
class KafkaConsumerManager {

  val props = new Properties()
  props.put("bootstrap.servers", KafkaServer.KAFKA_ADDRESS)
  props.put("group.id", "testGroup")
  props.put("enable.auto.commit", "true")
  props.put("auto.commit.interval.ms", "1000")
  props.put("linger.ms", "1")
  props.put("session.timeout.ms", "3000")
  props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer")
  props.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer")
  props.put("zookeeper.connect", KafkaServer.ZOOKEEPER_ADDRESS)

  val consumer = new KafkaConsumer[String, String](props)

  def start() {
    println("Start Consumer")
    consumer.subscribe(Arrays.asList("test-counter"))

    while (true) {
      val records = consumer.poll(100)

      val iterator = records.iterator()

      while (iterator.hasNext) {
        val record = iterator.next()
        printf("Consumer: offset = %d, key = %s, value = %s \n", record.offset(), record.key(), record.value())
      }
    }
  }
}
